File: backend-dotnetcore/ApiModels/MovieSummary.cs
using Microsoft.AspNetCore.Mvc.ModelBinding;
using Swashbuckle.AspNetCore.Annotations;

namespace RUG.WebEng.Movies.ApiModels;

public class MovieSummary
{
    [BindNever]
    [SwaggerSchema(ReadOnly = true)]
    public int Id { get; set; }
    public string Title { get; set; }
    public int? Year { get; set; }
    public string? Rating { get; set; }

    public static MovieSummary FromDatabase(Models.Movie movie) =>
        new()
        {
            Id = movie.Id,
            Title = movie.Title,
            Year = movie.Year,
            Rating = movie.Rating
        };
}

File: backend-express/api-models/movie-summary.ts
import { Movie as DbMovie } from "models/movie";

export class MovieSummary {
    id: number;
    title: string;
    year?: number;
    rating?: string;

    public static fromDatabase(movie : DbMovie) : MovieSummary {
        return {
            id: movie.id,
            title: movie.title,
            rating: movie.rating,
            year: movie.year
        };
    }
}